Tasks/PublishTestResultsV2/publishtestresults.ts

```diff
@@ -121,7 +124,11 @@ async function run() {
         }
 
         if (testResultsFilesCount === 0) {
-            tl.warning('No test result files matching ' + testResultsFiles + ' were found.');
+              if (failTaskOnMissingResultsFile) {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('NoMatchingFilesFound', testResultsFiles));
+            } else {
+                tl.warning(tl.loc('NoMatchingFilesFound', testResultsFiles));
```

Comment:
Just want to understand, this else part is a new behavior we'd be introducing I guess, which is not under the failTaskOnMissingResultsFile flag, what does this line do?

Reply:
Else part is the original behaviour till date which would be there by default.
see line no 124